src/eslint/rules/__tests__/no-extraneous.test.js

```diff
@@ -0,0 +1,70 @@
+import { RuleTester } from 'eslint';
+import rule from '../no-extraneous';
+
+RuleTester.setDefaultConfig({
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+    ecmaFeatures: { jsx: true },
+  },
+});
+
+var ruleTester = new RuleTester();
+ruleTester.run('no-extraneous', rule, {
+  valid: [
+    // it should ignore components/hooks with more than one occurrence
```

Comment:
i don't understand what that one does :)

Reply:
This makes sure you inject only things that are actually used in the component. If you have a long list of `di` and you change some functionality (removing code for instance) you might end up maintaining the imports are the injected dependency will still be marked as used by ESLint (used by `di(...)`).